### PR TITLE
solve CP2K parsing issue

### LINKED ISSUE #211: Read alpha and beta orbitals from cp2k unrestricted calculations
## New Feature
Parse the molecular orbitals for the CP2K unrestricted calculation printed in the log file

## The orbital log file
It has a structure like
```
 MO EIGENVALUES, MO OCCUPATION NUMBERS, AND SPHERICAL MO EIGENVECTORS AFTER SCF STEP -1

                              5                      6
                           0.0000000000000000     0.0000000000000000

                           1.0000000000000000     1.0000000000000000

     1     1  O  2s        0.2958563752067537    -0.3503316610611619
     2     1  O  3s       -0.0040795566634644     0.0560906457437746
....

 MO EIGENVALUES, MO OCCUPATION NUMBERS, AND SPHERICAL MO EIGENVECTORS AFTER SCF STEP -1

                              5                      6
                           0.0000000000000000     0.0000000000000000

                           1.0000000000000000     0.0000000000000000

     1     1  O  2s        0.1531315512503622     0.0000000000000000
     2     1  O  3s        0.0306583483779257     0.0000000000000000

....
 ALPHA MO EIGENVALUES, MO OCCUPATION NUMBERS, AND SPHERICAL MO EIGENVECTORS

                              5                      6
                          -0.3972464075508251    -0.1527841194681885

                           1.0000000000000000     1.0000000000000000

     1     1  O  2s       -0.3083673425711521     0.0000003538665076
     2     1  O  3s        0.0810440243240727    -0.0000001226930372
....
 BETA MO EIGENVALUES, MO OCCUPATION NUMBERS, AND SPHERICAL MO EIGENVECTORS

                              5                      6
                          -0.3243458939946837    -0.0588555206173679

                           1.0000000000000000     0.0000000000000000

     1     1  O  2s       -0.0000000010539389    -0.0000003505223712
     2     1  O  3s        0.0000001215174785     0.0000002348386832
....
```

### LINKED ISSUE #212: CP2K unordered molecular orbital coefficients
## The problem
CP2K prints the MO restart orbitals most of the times before the current point orbitals but on some few occasions it messes the restart coefficients with the MO of the current point.

## The solution
Throw away all restart orbitals.

### PR DESCRIPTION
**Pull request by felipeZ**
fix #211 and #212

### CODE REVIEW

File: src/qmflows/parsers/cp2KParser.py [hunk 1]
```diff
@@ -261,34 +275,49 @@ def remove_trailing(xs):
...
     properties = ["Number of occupied orbitals", "Number of molecular orbitals",
                   "Number of orbital functions"]
 
-    try:
-        xs = [fun_split(try_search_pattern(x, file_name)) for x in properties]  # type: ignore
-    except AttributeError as ex:  # try_search_pattern() can return None
-        raise RuntimeError(f"Failed to identify orbitals in {file_name!r}") from ex
+    meta = MO_metadata(*[fun_split(try_search_pattern(x, file_name)) for x in properties])
+
+    if any(getattr(meta, att) == 0 for att in ("nOrbitals", "nOccupied", "nOrbFuns")):
```
Comment (BvB93) [inline src/qmflows/parsers/cp2KParser.py:305]:
```suggestion
    if not all(meta[:-1]):
```
All elements are integers anyway, so `bool(i)` should be equivalent to `i == 0`.

File: src/qmflows/parsers/cp2KParser.py [hunk 2]
```diff
@@ -261,34 +275,49 @@ def remove_trailing(xs):
...
+        raise RuntimeError(f"Failed to identify orbitals in {file_name!r}")
+
+    return meta
 
-    return MO_metadata(*[int(x) for x in xs])
+
+def move_restart_coefficients_recursively(path: PathLike) -> None:
```
Comment (BvB93) [inline src/qmflows/parsers/cp2KParser.py:311]: So from what I understand this (recursivelly) splits a file into a bunch of other files until the block have interest has been successfully identified?

File: src/qmflows/parsers/cp2KParser.py [hunk 3]
```diff
@@ -261,34 +275,49 @@ def remove_trailing(xs):
...
     return added_mos, range_mos
 
 
 def read_cp2k_number_of_orbitals(file_name: PathLike) -> MO_metadata:
     """Look for the line ' Number of molecular orbitals:'."""
-    def fun_split(string: str) -> str:
-        return string.split()[-1]
+    def fun_split(string: Optional_[str]) -> int:
+        if string is not None:
+            return int(string.split()[-1])
```
Comment (BvB93) [inline src/qmflows/parsers/cp2KParser.py:296]:
```suggestion
            return int(string.rsplit(maxsplit=1)[-1])
```
Should be a bit faster with the `maxsplit` parameter specified.

**Review by BvB93 — COMMENTED** [review]
Looking pretty good and seems like a useful feature.

File: src/qmflows/parsers/cp2KParser.py (hunk 2 above)
Comment (felipeZ) [reply src/qmflows/parsers/cp2KParser.py:311]: yes, indeed due to some problems with the printing, CP2K may print the block in disorder but the import one is always the last one

File: src/qmflows/parsers/cp2KParser.py (hunk 1 above)
Comment (felipeZ) [reply src/qmflows/parsers/cp2KParser.py:305]: Right, thanks for the suggestion!

**Pull request merged**